### DISCUSSION TITLE
[Script request] Element Synapse

### DISCUSSION THREAD

**Discussion by CrazyWolf13** [community-scripts/ProxmoxVE#36] [Request script]
### Application Name

Element Synapse

### Website

https://github.com/element-hq/synapse

### Description

Synapse is an open source [Matrix](https://matrix.org/) homeserver implementation, written and maintained by [Element](https://element.io/). [Matrix](https://github.com/matrix-org) is the open standard for secure and interoperable real time communications. You can directly run and manage the source code in this repository, available under an AGPL license. There is no support provided from Element unless you have a subscription.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.

**Comment by Drizztfire** [comment]
Hi,
It seems that the script is available ...
Sorry if the question is dumb or i didn't search enough.

Is there a way to specify the server name at installation time ? Because it's impossible to change the server name after installation

htanks for your help !

**Reply by MickLesk** [reply]
Why should this be impossible? No server name is specified in the installation process, only the redirection to the internal IP. 

Check the file;
/etc/matrix-synapse/homeserver.yaml

**Reply by tremor021** [reply]
actually, what he says is true. .deb itself gives gui to enter both server name and opt out of telemetry. for that reason you cannot run it with $STD because the install will freeze:

```
echo "matrix-synapse-py3 matrix-synapse/server-name string matrix" | debconf-set-selections
echo "matrix-synapse-py3 matrix-synapse/report-stats boolean false" | debconf-set-selections
```
you also cannot entert it AFTER the installation because it gets hardcoded in the database

I'm gonna try and make a prompt for it during the installation, can't promise it will be ugly though

**Reply by MickLesk** [reply]
Use read -p prompt after or before a msg Block. (You can found here some Scripts With read -p)

**Reply by tremor021** [reply]
Pushed a update to the script. Should be able to enter server name before .deb installs

**Reply by Drizztfire** [reply]
@tremor021 It works perfectly !

Thanks a lot !

**Comment by knario** [comment]
Hello, sorry if this question has already been answered elsewhere. I couldn’t find more information.
I installed the script yesterday, and I don’t see how to log in with a username and password.
If I try to create a new username and password from an Android app, it doesn’t work.
Does the script come with a default username and password? Something like admin admin?
Thank you very much.

**Reply by tremor021** [reply]
Registration of new users is made through CLI and the default configuration has it disabled as its a security risk.

You have to manually enable it and choose what kind of user verification you want if you want to do it through matrix client. You should read the documentation:
https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#enable_registration

**Reply by knario** [reply]
Muchas gracias!!! Voy a revisar la documentación.

**Comment by kipavy** [comment]
Hello, would it be possible to add to the scripts optionnal automatic bridges installations ? Would be reaaally useful.
